app/models/spree/payment/processing.rb

```diff
@@ -25,6 +25,13 @@ def authorize!(return_url = nil)
         gateway_action(source, :authorize, :pend, return_url: return_url)
       end
 
+      def request_user_authorization
+        return unless authorization_action_required?
```

Comment:
we can move this deeper, and make each gateway determine if there's further action required.